vortex-array/src/arrays/chunked/compute/take.rs

```diff
@@ -24,26 +31,46 @@ impl TakeKernel for ChunkedVTable {
         let indices_mask = indices.validity_mask();
         let indices = indices.as_slice::<u64>();
 
-        let mut chunks = Vec::new();
+        let mut chunks: Vec<ArrayRef> = Vec::new();
         let mut indices_in_chunk = BufferMut::<u64>::empty();
         let mut start = 0;
         let mut stop = 0;
+        let mut builder: Option<Box<dyn ArrayBuilder>> = None;
+        let max_chunks_threshold = array.nchunks() * CHUNK_GROWTH_MULTIPLIER;
+
         // We assume indices are non-empty as it's handled in the top-level `take` function
         let mut prev_chunk_idx = array.find_chunk_idx(indices[0].try_into()?).0;
         for idx in indices {
             let idx = usize::try_from(*idx)?;
             let (chunk_idx, idx_in_chunk) = array.find_chunk_idx(idx);
 
             if chunk_idx != prev_chunk_idx {
-                // Start a new chunk
+                // Check if we're exceeding the threshold
+                if chunks.len() >= max_chunks_threshold && builder.is_none() {
+                    // Initialize builder and canonicalize all existing chunks
+                    let mut new_builder = builder_with_capacity(
+                        &array.dtype().clone().union_nullability(nullability),
```

Comment:
are you concerned about creating very long arrays. maybe we should try and keep the lengths of these compacted array a similar length to the original chunks.

Reply:
Why are very long arrays necessarily bad?